### PR TITLE
Adding Euclidean distance to utils 

### PR DESCRIPTION
**Pull request by JHLee0513**

### CODE REVIEW

File: include/utils/utils.hpp [hunk 1]
```diff
@@ -47,6 +48,15 @@ std::vector<T> linspace(T a, T b, std::size_t N) {
     return xs;
 }
 
+template <typename T>
+double euclidean(std::vector<T> a, std::vector<T> b) {
```
Comment (vinitha910) [inline include/utils/utils.hpp]: rename to `euclidean_distance`

**Comment by JHLee0513** [comment]
Added tests

File: tests/utils/test_euclidean_distance.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+/// Check that Euclidean distance with float is correct
+TEST(TestSuite, EuclideanFloatTest) {
+    std::vector<float> vector1{-16.75, 5.51, 0.49};
+    std::vector<float> vector2{100.0, 15.2, 0.00};
+    double result = utils::euclidean_distance(vector1, vector2);
+    EXPECT_NEAR(117.1525, result, 0.0001);
+}
+
+}  // namespace test
+}  // namspace utils
```
Comment (vinitha910) [inline tests/utils/test_euclidean_distance.cpp]: `namespace`

File: tests/utils/test_euclidean_distance.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    EXPECT_NEAR(107.7287, result, 0.0001);
+}
+
+/// Check that Euclidean distance with int is correct
+TEST(TestSuite, EuclideanIntTest) {
+    std::vector<int> vector1{10, 16, 5};
+    std::vector<int> vector2{100, 15, 0};
+    double result = utils::euclidean_distance(vector1, vector2);
+    EXPECT_NEAR(90.1443, result, 0.0001);
+    vector1[0] = -10;
```
Comment (vinitha910) [inline tests/utils/test_euclidean_distance.cpp]: `vector1 = {-10, -9, 1};`

File: tests/utils/test_euclidean_distance.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+namespace utils {
+namespace test {
+
+/// Check that Euclidean distance with double is correct
+TEST(TestSuite, EuclideanDoubleTest) {
+    std::vector<double> vector1{10.0, 16.4, 5.7};
+    std::vector<double> vector2{100.0, 15.2, 0.0};
+    double result = utils::euclidean_distance(vector1, vector2);
+    EXPECT_NEAR(90.1883, result, 0.0001);
+    vector1[0] = -5.5;
```
Comment (vinitha910) [inline tests/utils/test_euclidean_distance.cpp]: `vector1 = {-5.5, -6.6, 0}`

File: tests/utils/test_euclidean_distance.cpp (hunk 3 above)
Comment (vinitha910) [reply tests/utils/test_euclidean_distance.cpp]: Testing just one set of vectors is enough, you could just get rid of this second test

**Pull request merged**